src/components/Posts/PostDetails.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, useState, useEffect } from 'react';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../Comments/NewCommentForm';
+import { useCommentsContext } from '../../hooks/useCommentsContext';
+import { usePostsContext } from '../../hooks/usePostsContext';
+import { CommentList } from '../Comments/CommentList';
+import { Notification } from '../Notification';
+import { NotificationType } from '../../types';
+import { useGlobalContext } from '../../hooks/useGlobalContext';
+
+export const PostDetails: FC = () => {
+  const { error } = useGlobalContext();
+  const { selectedPost } = usePostsContext();
+  const { isCommentsLoading } = useCommentsContext();
+
+  const [writingNewPost, setWritingNewPost] = useState(false);
+
+  useEffect(() => {
+    setWritingNewPost(false);
+  }, [selectedPost]);
+
+  if (!selectedPost) {
```

Comment:
This condition only renders an error notification when a selected post does not exist, but it does not provide any fallback if the selected post is null or undefined. It would be good to handle these edge cases.